formulas/functions/reference.js

```diff
@@ -291,6 +291,71 @@ const ReferenceFunctions = {
 
   MATCH: () => {},
 
+  /**
+   * 
+   * @param {*} Origin : Required. The reference from which you want to base the offset. 
+   *                        Reference must refer to a cell or range of adjacent cells; otherwise, 
+   *                        OFFSET returns the #VALUE! error value.
+   * @param {*} Rows : Required. The number of rows, up or down, that you want the upper-left cell to refer to.
+   *                   Using 5 as the rows argument specifies that the upper-left cell in the reference is five rows below reference. 
+   *                   Rows can be positive (which means below the starting reference) or negative (which means above the starting reference).
+   * @param {*} Cols : Required. The number of columns, to the left or right, that you want the upper-left cell of the result to refer to. 
+   *                   Using 5 as the cols argument specifies that the upper-left cell in the reference is five columns to the right of reference. 
+   *                   Cols can be positive (which means to the right of the starting reference) or negative (which means to the left of the starting reference).
+   * @param {*} Height : Optional. The height, in number of rows, that you want the returned reference to be. Height must be a positive number.
+   * @param {*} Width : Optional. The width, in number of columns, that you want the returned reference to be. Width must be a positive number.
+   */
+  OFFSET: (context, origin, rows, cols, height = null, width = null) => {
+    if(!H.isCellRef(origin))
+      throw FormulaError.VALUE;
+      
+    const originRow = origin.ref.row
+    const originCol = origin.ref.col
+
+    rows = H.accept(rows, Types.NUMBER);
+    cols = H.accept(cols, Types.NUMBER);
+
+    const newRow = originRow + rows;
+    const newCol = originCol + cols;
+
+    if(newRow <= 0 || newCol <= 0)
+      throw FormulaError.REF;
+
+    if(height == null && width == null){
+      ref = {ref: { col: newCol, row: newRow}, isArray: false}
+      return context.utils.extractRefValue(ref).val
+    }
+    
+    if(height == null)
+      height = 0;
```

Comment:
According to Excel definition: https://support.microsoft.com/en-us/office/offset-function-c8de19ae-dd79-4b9b-a14e-b4d906d11b66

> If height or width is omitted, it is assumed to be the same height or width as reference.

This is missing from current implementation.